packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp

```diff
@@ -57,13 +57,115 @@ namespace DataTransferKit
 namespace UnitTest
 {
 //---------------------------------------------------------------------------//
-// Hex mesh constructor.
+// Num cells constructor.
+ReferenceHexMesh::ReferenceHexMesh( 
+    const Teuchos::RCP<const Teuchos::Comm<int> >& comm,
+    double x_min,
+    double x_max,
+    const int x_num_cells,
+    double y_min,
+    double y_max,
+    const int y_num_cells,
+    double z_min,
+    double z_max,
+    const int z_num_cells )
+{
+    Teuchos::Array<double> x_edges = 
+        buildEdgeArray( x_min, x_max, x_num_cells );
+    Teuchos::Array<double> y_edges = 
+        buildEdgeArray( y_min, y_max, y_num_cells );
+    Teuchos::Array<double> z_edges = 
+        buildEdgeArray( z_min, z_max, z_num_cells );
+    buildMesh( comm, x_edges, y_edges, z_edges );
+}
+
+//---------------------------------------------------------------------------//
+// Edge array constructor.
 ReferenceHexMesh::ReferenceHexMesh(
     const Teuchos::RCP<const Teuchos::Comm<int> >& comm,
     const Teuchos::Array<double>& x_edges,
     const Teuchos::Array<double>& y_edges,
     const Teuchos::Array<double>& z_edges)
 {
+    buildMesh( comm, x_edges, y_edges, z_edges );
+}
+
+//---------------------------------------------------------------------------//
+// Get the function space.
+Teuchos::RCP<DataTransferKit::FunctionSpace>
+ReferenceHexMesh::functionSpace() const
+{
+    return d_function_space;
+}
+
+//---------------------------------------------------------------------------//
+// Create a nodal field over the locally-owned nodes.
+Teuchos::RCP<DataTransferKit::Field>
+ReferenceHexMesh::nodalField( const int field_dim ) const
+{
+    // Select the locally-owned nodes.
+    DataTransferKit::LocalEntityPredicate pred(
+        d_function_space->entitySet()->communicator()->getRank() );    
+    auto node_it =
+        d_function_space->entitySet()->entityIterator( 0, pred.getFunction() );
+
+    // Extract the node ids.
+    auto nodes_begin = node_it.begin();
+    auto nodes_end = node_it.end();
+    Teuchos::Array<DataTransferKit::EntityId> node_ids;
+    node_ids.reserve( node_it.size() );
+    for ( node_it = nodes_begin; node_it != nodes_end; ++node_it )
+    {
+        node_ids.push_back( node_it->id() );
+    }
+
+    // Build the field.
+    Teuchos::ArrayRCP<double> field_data( node_it.size() * field_dim );
+    return Teuchos::rcp(
+        new DataTransferKit::EntityCenteredField(
+            node_ids(),
+            field_dim,
+            field_data,
+            DataTransferKit::EntityCenteredField::BLOCKED) );
+}
+
+//---------------------------------------------------------------------------//
+// Create a nodal field over the locally-owned + ghosted nodes.
+Teuchos::RCP<DataTransferKit::Field>
+ReferenceHexMesh::ghostedNodalField( const int field_dim ) const
```

Comment:
This is exactly the same as the `ReferenceHexMesh::getNodalFIeld` except that you'd pass a predicate "select all". Reuse if possible.